mat64/dense_arithmetic.go

```diff
@@ -641,9 +644,10 @@ func (m *Dense) Exp(a Matrix) {
 		scaling = 4
 	)
 
-	var small, power Dense
+	small := &Dense{}
 	small.Scale(math.Pow(2, -scaling), a)
-	power.Clone(&small)
+	power := &Dense{}
```

Comment:
The code is fine, but do you have guidelines as to when one should use
````
var a &Dense{}
````
as opposed to 
````
var a Dense
````
I don't feel like I've gotten the hang of it.